packages/website-eslint/build.ts

```diff
@@ -0,0 +1,139 @@
+    define: {
+      'process.env.NODE_ENV': '"production"',
+      'process.env.NODE_DEBUG': 'false',
+      'process.env.IGNORE_TEST_WIN32': 'true',
+      'process.env.DEBUG': 'false',
+      'process.emitWarning': 'console.warn',
+      'process.platform': '"browser"',
+      'process.env.TIMING': 'undefined',
+      'define.amd': 'false',
+      global: 'window',
+      // 'process.env': 'undefined',
+      // process: 'undefined',
+    },
+    alias: {
+      util: requireResolved('./mocks/util.js'),
+      assert: requireResolved('./mocks/assert.js'),
+      path: requireResolved('./mocks/path.js'),
+      typescript: requireResolved('./mocks/typescript.js'),
+      'lru-cache': requireResolved('./mocks/lru-cache.js'),
+    },
+    plugins: [
+      {
+        name: 'replace-plugin',
+        setup(build): void {
+          build.onLoad(
+            makeFilter([
+              '/eslint-utils/rule-tester/RuleTester.js',
+              '/utils/dist/ts-eslint/ESLint.js',
+              '/utils/dist/ts-eslint/RuleTester.js',
+              '/utils/dist/ts-eslint/CLIEngine.js',
+            ]),
+            async args => {
+              console.log('onLoad:replace', args.path);
+              const text = await requireMock('./mocks/empty.js');
+              return { contents: text, loader: 'js' };
+            },
+          );
+          build.onLoad(
+            makeFilter('/eslint/lib/unsupported-api.js'),
+            async args => {
+              console.log('onLoad:eslint:unsupported-api', args.path);
+              let text = await requireMock('./mocks/eslint-rules.js');
+              // this is needed to bypass system module resolver
+              text = text.replace('vt:eslint/rules', normalizePath(rulesPath));
+              return { contents: text, loader: 'js' };
+            },
+          );
+          build.onLoad(makeFilter('/eslint/lib/api.js'), async args => {
+            console.log('onLoad:eslint', args.path);
+            let text = await requireMock('./mocks/eslint.js');
+            // this is needed to bypass system module resolver
+            text = text.replace('vt:eslint/linter', normalizePath(linterPath));
+            return { contents: text, loader: 'js' };
+          });
+          build.onLoad(
+            makeFilter('/typescript-estree/dist/index.js'),
+            async args => {
+              console.log('onLoad:typescript-estree', args.path);
+              const text = await requireMock('./mocks/ts-estree.js');
+              return { contents: text, loader: 'js' };
+            },
+          );
+        },
+      },
+    ],
+  });
+
+  if (output.errors) {
+    for (const error of output.errors) {
+      console.error(error);
+    }
+    for (const warning of output.warnings) {
+      console.warn(warning);
+    }
+
+    if (output.errors.length > 0) {
+      throw new Error('error occurred');
+    }
+  }
+}
+
+console.time('building eslint for web');
+
+buildPackage('index', './src/index.js')
```

Comment:
instead of using rollup to bundle packages, i opted into using esbuild and generally cleaned up previous mess, 

as we are no longer mutating code without producing source maps, i noticed significat improvement to debugging in browser